interpreter/valid/valid.ml

```diff
@@ -155,6 +155,12 @@ let check_memop (c : context) (memop : 'a memop) get_sz at =
   require (1 lsl memop.align <= size) at
     "alignment must not be larger than natural"
 
+let check_atomic_memop (c : context) (memop : 'a memop) get_sz at =
```

Comment:
How about keeping the single `check_memop` and pass it an optional expected sharability `sh` and then check"
```
require (sh = None || sh = Some share) ...
```

Reply:
Done